### ISSUE TITLE
License on this?

### ISSUE THREAD

**Issue by NathanKell** [magico13/FlightSchool#1]
Fiiiiiinally getting back to that old thread on kerbal training. You mentioned there this would 'probably' be MIT. What is it now? I will need to make some changes for RP-0, and since you never released it I'd be the one to release now I guess (possibly as part of RP-0 itself?), but all that depends on What You Want (tm).

Cheers!

**Comment by magico13** [comment]
Yeah, let's go with MIT. This is definitely unfinished and I don't really even remember what it does or does not do, so feel free to do whatever you want with/to it. If you do end up using this as a base for something I definitely wouldn't mind being included in some credits somewhere ;)

**Comment by NathanKell** [comment]
Of *course* I'd include you, regardless, even if you did this as public domain. Yeesh. :P

I got it to compile, but indeed there's a lot that isn't hooked up yet. I'm probably (sorry) gonna chainsaw it up a bit just to suit our purposes rather than make the wonderfully generic useful-to-all version you'd make. :]

I'll also need to make a few changes to KCT, hope that's OK--I'll make sure to leave everything as disabled by default, then send in the PR.

(The minimalist version will just be, you need to have proficiency in a capsule and be trained in a mission before KCT will let you be added to that capsule during launch prep.)

**Comment by NathanKell** [comment]
I hooked up enough to make it add flightlog entries and extra XP. That should probably be enough for you if you ever want to resurrect or donate it.

I then went and imported it into RP-0 (to use our crew handler backend) and took a chainsaw to it to strip what we won't need. I'll make some changes to KCT tomorrow and hopefully it'll Just Work.
(And of course I added the credits, as they will be in the OP too :) )

Cheerio!

**Comment by magico13** [comment]
I'm sorry that you'll have to look at the mess that is the KCT code... If you're changing which crew are available when launching then you'll probably want to change that here: https://github.com/magico13/KCT/blob/development/Kerbal_Construction_Time/KCT_GUI.cs#L1305

Although if you're also adding something based on the part the Kerbal is being put into, that'll be more of a challenge. I'm sorry all that code is such a mess, I wrote it 3 years ago before I had any idea of what I was doing.

**Comment by NathanKell** [comment]
Hah!

I'm glad RSS's code no longer exists tbh, then I don't have to see it :D

Yep, looks like I'll need to pass around the part crew is being put into....

**Comment by magico13** [comment]
>Yep, looks like I'll need to pass around the part crew is being put into....

It might not be as bad as I first thought, but it's not going to be good. The part is known at each point (though it might be passed around stupidly, but you can get it) but you'll probably have to inject your code in a bunch of places because duplicating code is always better than breaking things into functions and separating UI code from business logic.

**Comment by NathanKell** [comment]
All done and pushed to the KSP-RO fork!
